src/Monitoring/ArrayMetricsAgent.php

```diff
@@ -9,36 +9,76 @@
 
 
 use DateTimeImmutable;
-use PHPUnit\Framework\Assert;
+use function array_filter;
+use function array_values;
 
 class ArrayMetricsAgent implements MetricsAgent
 {
-    protected array $timers = [];
```

Comment:
I guess very strictly speaking this is non-BC too, good example of advantage of `private` by default. OK with it given the context of the class, maybe mention in changelog just in case?

Reply:
I think originally we were thinking you could extend the ArrayMetricsAgent to support other types of handlers. It wasn't thought out very well to begin with